src/InitializeDatabaseScript.py

```diff
@@ -0,0 +1,201 @@
+    # queried.
+    def disconnect(self):
+        if self.db_server_conn:
+            self.db_cursor.close()
+            self.db_server_conn.close()
+            print("PostgreSQL connection is closed")
+
+    def create_database(self):
+        if self.db_server_conn:
+            try:
+                self.database_params = config.config('postgres_db')
+                self.db_server_conn.set_isolation_level(ISOLATION_LEVEL_AUTOCOMMIT)
+                database_name = self.database_params.get('database')
+                sqlCreateDatabase = "create database " + database_name + ""
+                self.db_cursor.execute(sqlCreateDatabase)
+
+            except (Exception, psycopg2.Error) as e:
+                self.debugLogFile.writeToFile("Could not create Database " + str(e))
+
+    def create_table(self):
+        # need to connect to the database that I am actually creating the table in.
+        conn = None
+        cursor = None
+        try:
+            # connect to the database on the postgres server.
+            conn = psycopg2.connect(**self.database_params)
+            # conn = self.database_connect()
+            # once we are connected to the database that we created on the postgreSQL server
+            # we can create a table in that database.
+            cursor = conn.cursor()
+            # self.bug_table_name =
+
+            # basic Table
+            table = ("CREATE TABLE " + self.bug_table_name + """ (
+                        bug_title text NULL,
+                        bug_description text NULL,
+                        bug_traceback_info text NULL,
+                        bug_resolved BOOL NULL,
+                        PRIMARY KEY (bug_title)
+                        ); """)
+            # execute the sql query
+            cursor.execute(table)
+            # commit the changes so they are saved
+            conn.commit()
+
+        except (Exception, psycopg2.Error) as e:
+            self.debugLogFile.writeToFile("Could not create table " + str(e))
+
+        finally:
+            # close the connection to the database to prevent memory leaks
+            if conn:
+                cursor.close()
+                conn.close()
+
+    def list_insert(self, bug_record):
+        conn = None
+        cursor = None
+        try:
+            conn = psycopg2.connect(**self.database_params)
+            cursor = conn.cursor()
+            # sql_values = []
+            # for item in bug_record:
+                # sql_values.append(item)
+            # insert_values = (bug_record[0], bug, "traceback info test", False)
+            insert_query = (" INSERT INTO " + self.bug_table_name + """(
+                            bug_title, 
+                            bug_description, 
+                            bug_traceback_info, 
+                            bug_resolved) 
+                            VALUES (%s, %s, %s, %s) """)
+
+            cursor.execute(insert_query, bug_record)
```

Comment:
Will this insert every item in the bug_record object into the correct column?

Reply:
Yeah it does. But it depends on the order they are passed in as an argument in the bug_record and the order they are listed in the query